lib/hci-socket/gatt.js

```diff
@@ -558,7 +558,11 @@ Gatt.prototype.write = function (serviceUuid, characteristicUuid, data, withoutR
       const opcode = data[0];
 
       if (opcode === ATT_OP_WRITE_RESP) {
-        this.emit('write', this._address, serviceUuid, characteristicUuid);
+        this.emit('write', this._address, serviceUuid, characteristicUuid, 0);
+      } else {
+        if (opcode === ATT_OP_ERROR && data.length == 5) {
```

Comment:
Mind explaining the length check here? Perhaps a comment would help, and could make `5` a constant.

Reply:
Sure will add comment. BLE Error response is 5 bytes. See section 3.4.1 .1 at https://iotexpert.com/2019/01/02/ble-write-request-write-command-signed-write-command-prepare-write/

For error response, opcode = ATT_OP_ERROR
// data[0]: opcode
// data[1]: request code
// data[2-3]: attribute handle
// data[4]: error code

The length check may not be needed b/c this should be checked in lower layers for a complete error response packet. I don't think there is harm keeping it.

Reply:
I'll just remove the length check; they don't check else where in noble.

Reply:
Got it, makes sense.